docs/pages/protocol/message-types.md

```diff
@@ -0,0 +1,100 @@
+# Message types with XMTP
+
+This document outlines the key message types that developers encounter when building with XMTP, explaining their purpose and when they occur. XMTP inherits these message types from its implementation of [Messaging Layer Security](https://www.rfc-editor.org/rfc/rfc9420.html)(MLS) to provide secure, end-to-end encrypted messaging.
+
+## Overview
+
+XMTP uses several message types to establish and maintain secure communication:
+
+- **Key package** - Cryptographic credentials for user installations
+- **Welcome** - Bootstraps new members into existing groups
+- **Commit** - Updates group state, membership, and permissions
+- **Identity update** - Verifies and authenticates user identities
```

Comment:
I don't think this is an official message type - but surface it here per XIP-69: https://community.xmtp.org/t/xip-69-message-retention-and-expiry-strategy-for-xmtpd/955

Not sure if helpful?

Reply:
This is so wonderful, @richardhuaaa - I've implemented your suggestions. I appreciate your care for and explanation of the details. So good.